src/ccl_power.c

```diff
@@ -402,6 +458,20 @@ static void ccl_cosmology_compute_power_class(ccl_cosmology * cosmo, int * statu
   struct file_content fc;
 
   ErrorMsg errmsg; // for error messages
+  
+  // Check if w!=-1 and halofit is enabled (raise error if so). CLASS uses a 
+  // spline to rescale the LambdaCDM halofit output if w!=-1, but this takes 
+  // an extremely long time if z_max_pk is set to a large value (which it is 
+  // by default in CCL). Instead of using a different z_max_pk (actually, 
+  // pk_eq_z_max) for different power spectrum methods, we just raise an error 
+  // for this method if w!=-1. 
```

Comment:
This seems like a serious restriction. Why can't we use Takahashi et al with CLASS here? Should we not rely on CLASS to compute halofit at all maybe? This might make out lives easier.

Reply:
That might be a good option. The CLASS version of halofit has some slightly hacky bits, like the spline rescaling that is causing these problems...